typescript/sdk/src/core/HyperlaneRelayer.ts

```diff
@@ -0,0 +1,221 @@
+import { TransactionReceipt } from '@ethersproject/providers';
+import { ethers } from 'ethers';
+import { Logger } from 'pino';
+import { z } from 'zod';
+
+import {
+  Address,
+  assert,
+  objMap,
+  objMerge,
+  pollAsync,
+  promiseObjAll,
+} from '@hyperlane-xyz/utils';
+
+import { DerivedHookConfig, EvmHookReader } from '../hook/EvmHookReader.js';
+import { HookConfigSchema } from '../hook/schemas.js';
+import { DerivedIsmConfig, EvmIsmReader } from '../ism/EvmIsmReader.js';
+import { BaseMetadataBuilder } from '../ism/metadata/builder.js';
+import { IsmConfigSchema } from '../ism/schemas.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { ChainName } from '../types.js';
+
+import { HyperlaneCore } from './HyperlaneCore.js';
+import { DispatchedMessage } from './types.js';
+
+const WithAddressSchema = z.object({
+  address: z.string(),
+});
+
+const DerivedHookConfigWithAddressSchema =
+  HookConfigSchema.and(WithAddressSchema);
+const DerivedIsmConfigWithAddressSchema =
+  IsmConfigSchema.and(WithAddressSchema);
+
+export const RelayerCacheSchema = z.object({
+  hook: z.record(z.record(DerivedHookConfigWithAddressSchema)),
+  ism: z.record(z.record(DerivedIsmConfigWithAddressSchema)),
+});
+
+type RelayerCache = z.infer<typeof RelayerCacheSchema>;
+
+export class HyperlaneRelayer {
+  private multiProvider: MultiProvider;
+  private metadataBuilder: BaseMetadataBuilder;
```

Comment:
There's pretty much no reason to make member args like these not `protected`, especially when the data is likely public in other places anyway

Reply:
imo `private` is about encapsulation rather than "privacy" but agree they can be protected